src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,72 @@
+package duke;
+
+import java.io.*;
+
+/**
+ * Deals with loading tasks from the file and saving tasks in the file
+ *
+ * @author sibinhho99-nus
+ */
+public class Storage {
+    public TaskList taskList = new TaskList();
+
+    /**
+     * Loads the task list from storage
+     */
+    public void load() throws IOException {
+        FileReader fr = new FileReader("storage.txt");
+        BufferedReader bufferedReader = new BufferedReader(fr);
+        while (bufferedReader.ready()) {
+            String line = bufferedReader.readLine();
+
+            if (line.charAt(3) == 'D') {
+                boolean isDone = line.charAt(6) == 'X';
+                String name = line.split(" \\(by: ")[0].substring(9);
+
+                String doneBy;
+                doneBy = line.split(" \\(by: ")[1];
+                doneBy = doneBy.substring(0, doneBy.length() - 1); // removes end bracket
+
+                taskList.add(new TaskDeadlines(isDone, name, doneBy));
+            } else if (line.charAt(3) == 'E') {
+                boolean isDone = line.charAt(6) == 'X';
+
+                String name = line.split(" \\(at: ")[0].substring(9);
+
+                String startAt;
+                startAt = line.split(" \\(at: ")[1];
+                startAt = startAt.substring(0, startAt.length() - 1); // removes end bracket
+
+                taskList.add(new TaskEvents(isDone, name, startAt));
+            } else { // T
+                boolean isDone = line.charAt(6) == 'X';
+                taskList.add(new TaskToDos(isDone, line.substring(9)));
+            }
```

Comment:
Instead of using magic numbers, perhaps "abstract" out the 3, 6, and 9 outside as static variables to make the program easier to understand?